fe/fe-core/src/main/java/com/starrocks/alter/LakeTableSchemaChangeJob.java

```diff
@@ -741,7 +741,12 @@ List<MaterializedIndex> visualiseShadowIndex(@NotNull LakeTable table) {
 
     @Override
     protected boolean cancelImpl(String errMsg) {
-        if (jobState == JobState.CANCELLED || jobState == JobState.FINISHED_REWRITING || jobState == JobState.FINISHED) {
+        if (jobState == JobState.CANCELLED || jobState == JobState.FINISHED) {
+            return false;
+        }
+
+        // Cancel a job of state `FINISHED_REWRITING` only when the database or table has been dropped.
+        if (jobState == JobState.FINISHED_REWRITING && tableExists()) {
```

Comment:
What's the meaning of FINISHED_REWRITING?

Reply:
https://github.com/StarRocks/starrocks/blob/aab02f79b0f1f0dfaa51119f3b9877e595d96d99/fe/fe-core/src/main/java/com/starrocks/alter/AlterJobV2.java#L73

Reply:
https://github.com/StarRocks/starrocks/blob/aab02f79b0f1f0dfaa51119f3b9877e595d96d99/fe/fe-core/src/main/java/com/starrocks/alter/LakeTableSchemaChangeJob.java#L424